### PR TITLE
removes CrdsValue.{signature,data} from public interface

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Working towards making changes to `CrdsValue` serialization/deserialization, which requires not exposing inner fields to public interface.

#### Summary of Changes
Removed `CrdsValue.{signature,data}` from public interface

### CODE REVIEW

File: gossip/src/crds_value.rs [hunk 1]
```diff
@@ -44,8 +44,8 @@ pub const MAX_EPOCH_SLOTS: EpochSlotsIndex = 255;
 #[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
 pub struct CrdsValue {
-    pub signature: Signature,
-    pub data: CrdsData,
+    signature: Signature,
+    data: CrdsData,
```
Comment (gregcusack) [inline gossip/src/crds_value.rs:48]: this is going to cause breaking changes for gossip crate users. any concerns about this?

**Review by gregcusack — COMMENTED** [review]
what are the planned serialization/deserialization changes that require this breaking change?

File: gossip/src/crds_value.rs (hunk 1 above)
Comment (behzadnouri) [reply gossip/src/crds_value.rs:48]: I think leaving these fields as public was an oversight and they shouldn't have been exposed to begin with.
I doubt that anyone relies on these implementation details, and even if they are, they can easily update their code or make a fork of the crate.

**Comment by behzadnouri** [comment]
> what are the planned serialization/deserialization changes that require this breaking change?

I have changes that hold on to the bincode serialized bytes of `CrdsValue` in order to avoid wasteful repeated serialization of the same value.

File: gossip/src/crds_value.rs (hunk 1 above)
Comment (gregcusack) [reply gossip/src/crds_value.rs:48]: ya should be easy enough to update

**Review by gregcusack — APPROVED** [review]
lgtm!

**Pull request merged**